spec/support/concerns/project_updater_shared_examples.rb

```diff
@@ -0,0 +1,71 @@
+shared_examples 'a ProjectUpdater' do
+  let(:project) { subject.project }
+  let(:create_subject) {
+    subject.save
+    subject.reload
+  }
+
+  it {
+    is_expected.to respond_to(:update_project_etag)
+    is_expected.to callback(:update_project_etag).after(:save)
+    is_expected.to callback(:update_project_etag).after(:destroy)
+  }
+
+  describe '#update_project_etag' do
+    let!(:original_project_etag) { project.etag }
+    it {
+      expect(create_subject).to be_truthy
+      expect(change_subject).to be_truthy
+      subject.update_project_etag
+      project.reload
+      subject.reload
+      last_project_audit = project.audits.last
+      last_subject_audit = subject.audits.last
```

Comment:
These last audit variables can be declared using `let` (if they're needed at all)